app/manifest.go

```diff
@@ -289,18 +306,281 @@ func (v *VersionSchema) AsOpenAPI3() (*openapi3.Components, error) {
+//
+//	"<group>/<version>.<kind>": {...},
+//	"<group>/<version>.<kind>List": {...},
+//	"<group>/<version>.spec": {...}, // ...etc. for all other resources
+//
+// If you wish to exclude a field from your kind's object, ensure that the field name begins with a `#`, which will be treated as a definition.
+// Definitions are included in the returned map as types, but are not included as fields (alongside "spec","status", etc.) in the kind object.
+//
+// It will error if the underlying schema cannot be parsed as valid openAPI.
+//
+//nolint:funlen
+func (v *VersionSchema) AsKubeOpenAPI(gvk schema.GroupVersionKind, ref common.ReferenceCallback) (map[string]common.OpenAPIDefinition, error) {
+	// Convert the kin-openapi to kube-openapi
+	oapi, err := v.AsOpenAPI3()
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]common.OpenAPIDefinition)
 
-type ManifestOperatorInfo struct {
-	URL      string                             `json:"url" yaml:"url"`
-	Webhooks *ManifestOperatorWebhookProperties `json:"webhooks,omitempty" yaml:"webhooks,omitempty"`
+	kindProp := spec.Schema{
+		SchemaProps: spec.SchemaProps{
+			Description: "Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds",
+			Type:        []string{"string"},
+			Format:      "",
+		},
+	}
+	apiVersionProp := spec.Schema{
+		SchemaProps: spec.SchemaProps{
+			Description: "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources",
+			Type:        []string{"string"},
+			Format:      "",
+		},
+	}
+
+	// kind must always be present, partially declare it here so we can add subresources and dependencies as we iterate through them
+	kind := common.OpenAPIDefinition{
+		Schema: spec.Schema{
+			SchemaProps: spec.SchemaProps{
+				Type: []string{"object"},
+				Properties: map[string]spec.Schema{
+					"kind":       kindProp,
+					"apiVersion": apiVersionProp,
+					"metadata": {
+						SchemaProps: spec.SchemaProps{
+							Default: map[string]any{},
+							Ref:     ref("k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta"),
+						},
+					},
+				},
+			},
+		},
+		Dependencies: make([]string, 0),
+	}
+
+	// For each schema, create an entry in the result
+	for k, s := range oapi.Schemas {
+		sch, deps := oapi3SchemaToKubeSchema(s, ref, gvk)
+		// sort dependencies for consistent output
+		slices.Sort(deps)
+		result[fmt.Sprintf("%s/%s.%s", gvk.Group, gvk.Version, k)] = common.OpenAPIDefinition{
```

Comment:
very nit: consider declaring `fmt.Sprintf("%s/%s.%s", gvk.Group, gvk.Version, k)` separately as I can see it's repeated some times in this for loop